cdap-ui/app/cdap/components/Cloud/Profiles/Store/ActionCreator.js

```diff
@@ -133,10 +136,14 @@ export const fetchAggregateProfileMetrics = (namespace, profile, extraTags) => {
             if (!metricsMap.hasOwnProperty(query)) {
               metricsMap[query] = {};
             }
-            metricsMap[query]= {
-              ...metricsMap[query],
-              [metricName]: metricValue
-            };
+            if (metricsMap[query][metricName] !== '--') {
+              metricsMap[query][metricName] += metricValue;
```

Comment:
Since it's possible for `metricsMap[query] = {}`, it's possible `metricsMap[query][metricName]` is `undefined` right? There would be a JS error in this case?

Reply:
So line 113 has 

```
let metricsMap = {
  [ONEDAYMETRICKEY]: {
    runs: '--',
    minutes: '--'
  },
  [OVERALLMETRICKEY]: {
    runs: '--',
    minutes: '--'
  }
};
```

So here `query == ONEDAYMETRICKEY || OVERALLMETRICKEY` and `metricName == runs|minutes`
So it should have only one value `--` before we receive any metrics from the backend.

Reply:
Yeah.